docs/tutorial/flask.rst

```diff
@@ -0,0 +1,530 @@
+
+    If you changed the ``name`` or ``version`` in ``rockcraft.yaml`` or are
+    not on an ``amd64`` platform, the name of the ``.rock`` file will be
+    different for you.
+
+The rock needs to be copied to the MicroK8s registry so that it can be
+deployed in the Kubernetes cluster:
+
+.. literalinclude:: code/flask/task.yaml
+    :language: bash
+    :start-after: [docs:skopeo-copy]
+    :end-before: [docs:skopeo-copy-end]
+    :dedent: 2
+
+.. seealso::
+
+    See more: `skopeo <https://manpages.ubuntu.com/manpages/jammy/man1/skopeo.1.html>`_
+
+Create the charm
+================
+
+Create a new directory for the charm and go inside it:
+
+.. literalinclude:: code/flask/task.yaml
+    :language: bash
+    :start-after: [docs:create-charm-dir]
+    :end-before: [docs:create-charm-dir-end]
+    :dedent: 2
+
+We'll need a ``charmcraft.yaml``, ``requirements.txt`` and source code for
+the charm. The source code contains the logic required to operate the Flask
+application. Charmcraft will automate the creation of these files by using
+the ``flask-framework`` profile:
+
+.. literalinclude:: code/flask/task.yaml
+    :language: bash
+    :start-after: [docs:charm-init]
+    :end-before: [docs:charm-init-end]
+    :dedent: 2
+
+The files will automatically be created in your working directory.
+Pack the charm:
+
+.. literalinclude:: code/flask/task.yaml
+    :language: bash
+    :start-after: [docs:charm-pack]
+    :end-before: [docs:charm-pack-end]
+    :dedent: 2
+
+.. note::
+
+    Depending on your system and network, this step can take a couple
+    of minutes to finish.
+
+Once Charmcraft has finished packing the charm, you'll find a new file in your
+working directory with the ``.charm`` extension:
+
+.. literalinclude:: code/flask/task.yaml
+    :language: bash
+    :start-after: [docs:ls-charm]
+    :end-before: [docs:ls-charm-end]
+    :dedent: 2
+
+.. note::
+
+    If you changed the name in charmcraft.yaml or are not on the amd64 platform,
+    the name of the ``.charm`` file will be different for you.
+
+Deploy the Flask application
+============================
+
+A Juju model is needed to deploy the application. Let's create a new model:
+
+.. literalinclude:: code/flask/task.yaml
+    :language: bash
+    :start-after: [docs:add-juju-model]
+    :end-before: [docs:add-juju-model-end]
+    :dedent: 2
+
+.. warning::
+
+    If you are not on a host with the amd64 architecture, you will need to include
+    a constraint to the Juju model to specify your architecture. For example, for
+    the arm64 architecture, use
+    ``juju set-model-constraints -m flask-hello-world arch=arm64``.
+    Check the architecture of your system using ``dpkg --print-architecture``.
+
+Now the Flask application can be deployed using `Juju <https://juju.is/docs/juju>`_:
+
+.. literalinclude:: code/flask/task.yaml
+    :language: bash
+    :start-after: [docs:deploy-juju-model]
+    :end-before: [docs:deploy-juju-model-end]
+    :dedent: 2
+
+.. note::
```

Comment:
This one might be better off as a tip. Or – you know what's coming next – as a plain paragraph.

Reply:
Sounds good to me